player/models.py

```diff
@@ -142,12 +148,28 @@ def select_random_music(self):
         a = count / float(5)  # Le point où ca commence à monter
         b = count / float(27)  # La vitesse à laquelle ca monte
         x = random.uniform(1, count - a - 1)
-        i = min(int(math.floor(x + a - a * math.exp(-x / b))), len(musics) - 1)  # Can't select out of range music
+
+        i = min(int(math.floor(x + a - a * math.exp(-x / b))), musics.count() - 1)  # Can't select out of range music
         if i < 0:
             i = 0  # Can't get negative index
 
         return musics[i]
 
+    def fill_shuffle_playlist(self):
+        if self.shuffle:
+            current_shuffle_count = PlaylistTrack.objects.filter(room=self, track_type=PlaylistTrack.SHUFFLE).count()
+            while current_shuffle_count < self.nb_shuffle_items:
+                shuffled_track = self.select_random_music()
+                PlaylistTrack.objects.create(room=self, track=shuffled_track, track_type=PlaylistTrack.SHUFFLE)
```

Comment:
Is there a difference with `PlaylistTrack(room=self, track=shuffled_track, track_type=PlaylistTrack.SHUFFLE).save()` ?